files/initramfs-tools/arista-convertfs

```diff
@@ -0,0 +1,170 @@
+#!/bin/sh
+
+case $1 in
+    prereqs)
+        exit 0
+        ;;
+esac
+
+set -e
+# set -x
+total_mem=$(free | awk '/^Mem:/{print $2}')
+tmpfs_size=$(( $total_mem / 20 *  17 ))
+free_mem_thres=$(( $total_mem / 20 * 18 ))
+tmp_mnt='/mnt/ramdisk-convfs'
+root_mnt='/mnt/root-convfs'
+root_dev=''
+flash_dev=''
+block_flash=''
+aboot_flag=''
+backup_file=''
+
+# Get the fullpath of flash device, e.g., /dev/sda
+get_flash_dev() {
+    for dev in $(ls /sys/block); do
+        local is_mmc=$(echo "$dev" | grep 'mmcblk.*boot.*' | cat)
+        if [ -n "$is_mmc" ]; then
+            continue
+        fi
+        local devid=$(realpath "/sys/block/$dev/device")
+        local is_device=$(echo "$devid" | grep '^/sys/devices/' | cat)
+        local is_flash=$(echo "$devid" | grep "$block_flash" | cat)
+        if [ -n "$is_device" -a -n "$is_flash" ]; then
+            flash_dev="/dev/$dev"
+            return 0
+        fi
+    done
+    return 1
+}
+
+# Wait for root_dev to be ready
+wait_for_root_dev() {
+    local try_rounds=30
+    while [ $try_rounds -gt 0 ]; do
+        if [ -e "$root_dev" ]; then
+            return 0
+        fi
+        sleep 1
+        try_rounds=$(( $try_rounds - 1 ))
+    done
+    return 1
+}
+
+# Alway run cleanup before exit
+cleanup() {
+    if grep -q "$root_mnt" /proc/mounts; then
+        umount "$root_mnt"
+    fi
+    if grep -q "$tmp_mnt" /proc/mounts; then
+        umount "$tmp_mnt"
+    fi
+    [ -e "$root_mnt" ] && rmdir "$root_mnt"
+    [ -e "$tmp_mnt" ] && rmdir "$tmp_mnt"
+}
+trap cleanup EXIT
+
+notification() {
+cat << EOF
+A failure happend in modifying the root file system which stopped the upgrade. Manual interventions are needed to fix the issue. Note that:
+1) files in the old root file system may have been lost and the old partition table may have been corrupted;
+2) The files in the old root file system were copied to $tmp_mnt;
+3) The old partition table was dumped to the file $tmp_mnt/$backup_file by sfdisk;
+4) Quitting the current shell will lose all files mentioned above permanently.
+EOF
+}
+
+run_cmd() {
+    if ! eval "$1"; then
+        echo "$2"
+        notification
+        sh
+        exit 1
+    fi
+}
+
+# Extract kernel parameters
+set -- $(cat /proc/cmdline)
```

Comment:
set is parsing <name>=<value> from a string. 
The -- is to stop the option argument parsing for safeness. 
You can see the difference from the examples below.
grep --help | grep -q
grep --help | grep -- -q